### PR TITLE
language based filtering for subscription

### PR DESCRIPTION
**Pull request by chung-lai**
* Filter subscription based on preferredLanguage instead of Locale.current.languageCode. (languageCode only return app's localized language so in our case, it will always be `en`, `de` and `fr`)
* Add PremiumEligible logging to identify user eligibility

**Test:**

Change device language and not seeing the plan screen when it's not in our whitelist

### How was this tested?
- [x] I tested this manually
- [ ] I added automated tests
- [ ] Existing automated tests are enough to cover my changes

### CODE REVIEW

**Review by jedireza — APPROVED** [review]

File: Client/Frontend/Premium/PremiumStore.swift [hunk 1]
```diff
@@ -65,10 +65,19 @@ class PremiumStore: ObservableObject {
...
+            // preferredLanguage returns in format en, de-US...
+            let components = preferredLanguages[0].components(separatedBy: "-")
+            if components.count > 0 && PremiumStore.languages.contains(components[0]) {
                 return true
             }
+        } else {
+            // will offer premium in the case there is no preferredLanguage
+            // we want to track how frequent this case happens
+            ClientLogger.shared.logCounter(.NoPreferredLanguage)
+            return true
```
Comment (jedireza) [inline Client/Frontend/Premium/PremiumStore.swift:80]: I’m not sure we should return true here. With this, the function always returns true.

Comment (chung-lai) [reply Client/Frontend/Premium/PremiumStore.swift:80]: right, in the case when we don't see any preferred language, we consider user as eligible for now. Again, this case should be rare and we will be looking to see how often it happens.

Comment (chung-lai) [reply Client/Frontend/Premium/PremiumStore.swift]: but i do see there can be confusion. I just reversed the logic so it's more obvious :)

**Review by jedireza — APPROVED** [review]
Thanks for working through this!

**Pull request merged**